scripts/make_statistics.py

```diff
@@ -0,0 +1,539 @@
+import numpy as np
+import pandas as pd
+import pypsa
+import xarray as xr
+from _helpers import mock_snakemake, sets_path_to_root, to_csv_nafix
+from build_test_configs import create_test_config
+from ruamel.yaml import YAML
+from shapely.validation import make_valid
+
+
+def _multi_index_scen(rulename, keys):
+    return pd.MultiIndex.from_product([[rulename], keys], names=["rule", "key"])
+
+
+def _mock_snakemake(rule, **kwargs):
+    os.chdir(os.path.dirname(os.path.abspath(__file__)))
+    snakemake = mock_snakemake(rule, **kwargs)
+    sets_path_to_root("pypsa-earth")
+    return snakemake
+
+
+def generate_scenario_by_country(
+    path_base, country_list, out_dir="configs/scenarios", pre="config."
+):
+    """
+    Utility function to create copies of a standard yaml file available in path_base
+    for every country in country_list.
+    Copies are saved into the output directory out_dir
+
+    Note:
+    - the clusters are automatically modified for selected countries with limited data
+    - for landlocked countries, offwind technologies are removed (solar, onwind and hydro are forced)
+
+    Parameters
+    ----------
+        path_base : str
+            Path to the standard yaml file used as default
+        country_list : list
+            List of countries.
+            Note: the input is parsed using download_osm_data.create_country_list
+        out_dir : str (optional)
+            Output directory where output configuration files are executed
+    """
+
+    from _helpers import create_country_list, three_2_two_digits_country
+
+    clean_country_list = create_country_list(country_list)
+
+    df_landlocked = pd.read_csv(
+        "https://raw.githubusercontent.com/openclimatedata/countrygroups/main/data/lldc.csv"
+    )
+    df_landlocked["countries"] = df_landlocked.Code.map(three_2_two_digits_country)
+
+    n_clusters = {
+        "MG": 3,  # Africa
+        "BF": 1,
+        "BI": 1,
+        "BJ": 2,
+        "DJ": 1,
+        "GM": 2,
+        "LR": 2,
+        "LS": 3,
+        "SL": 1,
+        "TG": 1,
+        "CG": 2,
+        "GN": 3,
+        "SS": 1,
+        "ML": 1,
+        "TD": 2,
+        "CF": 1,
+        "ER": 1,  # Africa
+        "GF": 3,  # South America
+        "SR": 1,  # South America
+        "SG": 1,  # Asia
+        "FJ": 4,  # Oceania
+    }
+
+    for c in clean_country_list:
+        modify_dict = {"countries": [c]}
+
+        if c in n_clusters.keys():
+            modify_dict["scenario"] = {"clusters": [str(n_clusters[c])]}
+
+        if df_landlocked["countries"].str.contains(c).any():
+            modify_dict["electricity"] = {
+                "renewable_carriers": ["solar", "onwind", "hydro"]
+            }
+
+        create_test_config(path_base, modify_dict, f"{out_dir}/{pre}{c}.yaml")
+
+
+def collect_basic_osm_stats(path, rulename, header):
+    """
+    Collect basic statistics on OSM data: number of items
+    """
+
+    if os.path.exists(path) and (os.stat(path).st_size > 0):
```

Comment:
An pathlib alternative for this line:
`if Path("path/to/my/file").stat().st_size > 0:`
https://stackoverflow.com/a/55949699/13573820

Reply:
Done